### DISCUSSION TITLE
Is it possible to manage yazi plugins using lazy.nvim?

### DISCUSSION THREAD

**Discussion by mikavilpas** [mikavilpas/yazi.nvim#84] [Ideas]
As a neovim and lazy.nvim user, I update my lazy.nvim plugins almost daily. The experience is using it is very good. I get access to features such as:

- visually see which plugins will be updated before I make a choice of updating or not updating
- see which commits are included in the new version
- `<leader>l` (lowercase L) opens lazygit in the same ui seamlessly for browsing commit diffs
- `<leader>t` opens a floating terminal in the repository of the package in a similar manner
- local plugin directories can easily be set, so that I can load the plugins from my `~/git/` repository directory. useful when developing!
- versions of online plugins can be pinned or set to a specific git branch
- many git providers are supported out of the box, although I only use github and local directories
- lockfile support is very good (`lazy-lock.json`), and can be committed to my dotfiles repo, and optionally reverted back to (in case there are issues, just roll back. easy.)
- custom build commands can be used for packages that need to e.g. build extensions native to the system and processor architecture (most commonly tree-sitter parsers)

...there's probably tons more. Basically the feature set is insanely good.

My motivation: I like using this so much that I don't necessarily want to learn another system to manage yazi plugins. I want to use this system.

**Comment by mikavilpas** [comment]
I actually built a working proof of concept for this today. It's able to `git clone` a new yazi package to `/Users/mikavilpas/.local/share/nvim/lazy/fg.yazi`, and then create a symbolic link to my `yazi/plugins/` directory.

https://github.com/mikavilpas/yazi.nvim/assets/300791/e746ba3a-7606-428c-9e6b-a6cb05094930

Neovim does not load the plugin - it only allows managing its version using the lazy.nvim ui.

@sxyazi I was wondering if you have some upcoming plans for the plugin system that might conflict with this idea. I think this would require some kind of tinkering, so it would only be for "advanced users". Not really sure where I want to go with this idea yet.

**Comment by sxyazi** [comment]
Wow that's really cool!

---

> I was wondering if you have some upcoming plans for the plugin system that might conflict with this idea. I think this would require some kind of tinkering, so it would only be for "advanced users". Not really sure where I want to go with this idea yet.

Yeah, I have plans to improve the package manager in the future, right now it’s just an MVP. It’s not a priority, this means I won’t be spending time on it right away unless it runs into bugs and doesn’t have a simple workaround.

**Comment by mikavilpas** [comment]
The plugin manager integration feature with lazy.nvim has now been released. I think I managed to strike a reasonable balance between the spirit of lazy.nvim (the user is in control) and helping the user set the plugins up.

In this approach, lazy.nvim handles the plugin installation and updating, and yazi.nvim only helps with linking (symlinking) the plugin. I'm hoping this "helping step" is flexible enough that any future yazi plugin manager features could be supported there.

**Comment by mikavilpas** [comment]
I noticed there are plans in neovim and lazy.nvim to move to a new, ecosystem agnostic package specification:
- https://github.com/neovim/neovim/pull/25070
- https://github.com/folke/lazy.nvim/pull/910
- https://github.com/neovim/packspec/issues/41

I will be following the progress on this initiative. But for now it seems that there is no implementation to try out.

**Comment by mikavilpas** [comment]
I started a discussion demonstrating the general idea here, asking for feedback
https://github.com/folke/lazy.nvim/discussions/1488

**Comment by jrcasey** [comment]
Would you mind please adding an 'idiot's guide' to installing yazi with lazy?

**Reply by mikavilpas** [reply]
Maybe my config can serve as an example 

https://github.com/mikavilpas/dotfiles/blob/37e72f36c34e44e97d88ed02361b8ca82e1c04cc/.config/nvim/lua/plugins/my-file-manager.lua

Is there something specific you're struggling with?
